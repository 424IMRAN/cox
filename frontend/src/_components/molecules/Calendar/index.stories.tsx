import { Story, Meta } from '@storybook/react'
import { ThemeProvider } from '@material-ui/core/styles'
import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date'
import Calendar from '.'
import baseTheme from '../../../core-utils/theme'

export default {
  title: 'molecules/Calendar',
  component: Calendar,
  argTypes: {},
} as Meta

const Template: Story = ({ ...args }) => (
  <ThemeProvider theme={baseTheme}>
    <Calendar
      date={new Date()}
      onDateChange={(date: MaterialUiPickersDate) => {
        console.log(date)
      }}
      {...args}
    />
  </ThemeProvider>
)

export const DefaultCalendar = Template.bind({})
DefaultCalendar.args = {
  date: new Date(),
  onDateChange: (date: MaterialUiPickersDate): void => {
    console.log(date)
  },
}
