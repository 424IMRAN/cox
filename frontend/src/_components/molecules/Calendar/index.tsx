import React from 'react'
import DateFnsUtils from '@date-io/date-fns'
import PropTypes from 'prop-types'
import { DatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers'
import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date'
import { makeStyles } from '@material-ui/core'

type CalendarProps = {
  date: Date
  onDateChange: (date: MaterialUiPickersDate) => void
  [restProps: string]: unknown
}

const useStyles = makeStyles({
  root: {
    '.MuiTypography-root': {
      fontWeight: 400,
    },
  },
})
const Calendar = ({ date, onDateChange, ...restProps }: CalendarProps) => {
  const classes = useStyles()
  return (
    <>
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <DatePicker
          className={classes.root}
          autoOk
          variant="static"
          openTo="date"
          value={date}
          onChange={onDateChange}
          disableToolbar
          {...restProps}
          style={{ fontWeight: 400 }}
        />
      </MuiPickersUtilsProvider>
    </>
  )
}

Calendar.propTypes = {
  date: PropTypes.instanceOf(Date),
  onDateChange: PropTypes.func,
}
Calendar.defaultProps = {
  date: new Date(),
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onDateChange: () => {},
}

export default Calendar
