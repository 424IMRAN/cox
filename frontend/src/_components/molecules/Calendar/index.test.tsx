import React from 'react'
import { mount, shallow } from 'enzyme'
import { DatePicker } from '@material-ui/pickers'
import Calendar from '.'

describe('Text Field  Component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <Calendar
        date={new Date('August 19, 1975 23:15:30')}
        onDateChange={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Allows us to set props', () => {
    const wrapper = mount(
      <Calendar date={new Date()} onDateChange={jest.fn()} />
    )
    // expect(wrapper.props().date).toEqual(new Date())
    expect(wrapper.props().date.getDay()).toEqual(new Date().getDay())
  })

  it('onChange works', () => {
    const mockHandleChange = jest.fn()
    const component = mount(
      <Calendar date={new Date()} onDateChange={mockHandleChange} />
    )
    component
      ?.find(DatePicker)
      ?.props()
      ?.onChange?.(new Date('August 19, 1975 23:15:30') as any)
    expect(mockHandleChange).toHaveBeenCalledWith(
      new Date('August 19, 1975 23:15:30')
    )
  })
})
