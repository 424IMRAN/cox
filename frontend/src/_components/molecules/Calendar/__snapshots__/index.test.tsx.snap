// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Field  Component Renders correctly 1`] = `
<Fragment>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <PickerWithState
      allowKeyboardControl={true}
      autoOk={true}
      className="makeStyles-root-1"
      disableToolbar={true}
      invalidDateMessage="Invalid Date Format"
      maxDate={2100-01-01T00:00:00.000Z}
      maxDateMessage="Date should not be after maximal date"
      minDate={1900-01-01T00:00:00.000Z}
      minDateMessage="Date should not be before minimal date"
      onChange={[MockFunction]}
      openTo="date"
      style={
        Object {
          "fontWeight": 400,
        }
      }
      value={1975-08-19T17:45:30.000Z}
      variant="static"
      views={
        Array [
          "year",
          "date",
        ]
      }
    />
  </MuiPickersUtilsProvider>
</Fragment>
`;
